src/ucp/api/ucp.h

```diff
@@ -1172,6 +1224,73 @@ typedef struct ucp_worker_params {
 } ucp_worker_params_t;
 
 
+/**
+ * @ingroup UCP_ENDPOINT
+ * @brief UCP endpoint performance evaluation request attributes.
+ *
+ * The structure defines the attributes which characterize
+ * the request for performance estimation of a particular endpoint.
+ */
+typedef struct {
+    /**
+     * Mask of valid fields in this structure, using bits from
+     * @ref ucp_ep_perf_param_field_t.
+     * Fields not specified in this mask will be ignored.
+     * Provides ABI compatibility with respect to adding new fields.
+     */
+    uint64_t          field_mask;
+
+    /**
+     * Message size to use for determining performance.
+     * This field must be initialized by the caller.
+     */
+    size_t            message_size;
```

Comment:
I have question about "must be initialized by the caller". Does caller *must* initialized the field or we can leave it empty (aka corresponding field_mask is not set?)

Reply:
"must be initialized by the caller" means that the field must be initialized, and the corresponding field in field_mask should be set. Do we need to clarify it in the doc?

Reply:
Looking at implementation - if you don't sent the field, it does not cause any failure. So it appears that you don't *must* to set the field, it just does not return any useful information.

Reply:
you are right.
@gleon99 we need to fail with invalid param if message_size not passed

Reply:
Fixed